ocr.pm

```diff
@@ -1,31 +1,121 @@
 # Copyright 2009-2013 Bernhard M. Wiedemann
-# Copyright 2012-2020 SUSE LLC
+# Copyright 2012-2023 SUSE LLC
 # SPDX-License-Identifier: GPL-2.0-or-later
 
+=encoding utf8
+
+=head1 OCR Package
+
+Providing optical character recognition (OCR) functionality
+
+=cut
+
 package ocr;
 use Mojo::Base -strict, -signatures;
 require IPC::System::Simple;
+use autodie ':all';
+use File::Temp 'tempdir';
+use File::Which;
+use Scalar::Util 'blessed';
+use Exporter 'import';
+use bmwqemu 'fctwarn';
+
+our @EXPORT = qw(img_to_str ocr_installed);
+our $OCR_INSTALLED = which('gocr');
+
+=head2 img_to_str
+
+Reduces the information carried by the provided image to a character string.
+Probably the underlying implementation used by this function is not complete
+and might lead to wrong or incomplete results in some cases.
+
+=head3 Params
+
+=over
+
+=item *
+
+img - A tinycv image
+
+=back
+
+=head3 returns
+
+Character string of characters '-_.:,;()[]{}<>0-9a-zA-Z~+*^@#$%&/=?!§',
+where '§' is placed for each unrecognized character. Implicitly, '§'
+itself can't be recognized. Keep that in mind for comparison!
+Returns undef on failure.
+
+=cut
+
+sub img_to_str ($img) {
+    return unless $OCR_INSTALLED;
+    if (blessed($img)) {
+        unless ($img->isa('tinycv::Image')) {
+            fctwarn('Not a reference to object with type tinycv::Image!');
+            return;
+        }
+    }
+    else {
+        fctwarn('Not a reference to object with type tinycv::Image!');
+        return;
+    }
```

Comment:
Those two cases can actually be shrunk into one:
```suggestion
    if (!blessed($img) || !$img->isa('tinycv::Image')) {
        fctwarn('Not a reference to object with type tinycv::Image (internal error, probably a bug)!');
        return undef;
    }
```

This error message is likely still useless to a user. I'm not sure whether it can be improved, though.

Reply:
I have implemented the suggestion.